### PR TITLE
Fixes #2333. Representation type can be a subtype of a supertype

### LINKED ISSUE #2333: [Extension type] Add/update tests regarding subtype relations
The spec change is https://github.com/dart-lang/language/commit/c214fd960db1b3b4abe15981bbf7d657c3c29bfa

### PR DESCRIPTION
**Pull request by sgrekhov**

### CODE REVIEW

File: LanguageFeatures/Extension-types/superinterfaces_of_extension_type_A04_t09.dart [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+// SharedOptions=--enable-experiment=inline-class
+
+import "../../Utils/expect.dart";
+
+extension type V1<T>(T _) {}
+
+extension type V2<T extends num>(T _) implements V1<num> {}
+
+extension type ET<T extends V2<int>>(T id) implements V1<int> {}
```
Comment (eernstg) [inline LanguageFeatures/Extension-types/superinterfaces_of_extension_type_A04_t09.dart:30]: Maybe?:
```suggestion
extension type ET<T extends V2<int>>(T id) implements V1<num> {}
```
As far as I can see, we can't prove `T <: V1<int>`.

File: LanguageFeatures/Extension-types/superinterfaces_of_extension_type_A04_t10.dart [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+/// ...
+/// Assume that an extension type declaration DV named Name has representation
+/// type R, and that the extension type V1 with declaration DV1 is a
+/// superinterface of DV (note that V1 may have some actual type arguments).
+/// Assume that S is the instantiated representation type corresponding to V1. A
+/// compile-time error occurs if R is is neither a subtype of S nor a subtype of
+/// V1.
+///
+/// @description Checks that it is a compile-time error if an extension type has
+/// a representation type `R` and a superinterface with a representation type
```
Comment (eernstg) [inline LanguageFeatures/Extension-types/superinterfaces_of_extension_type_A04_t10.dart:17]:
```suggestion
/// a representation type `R` and a superinterface `V1` with a representation type
```

File: LanguageFeatures/Extension-types/superinterfaces_of_extension_type_A04_t10.dart [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+/// `S` and `R` is neither a subtype of `S` nor a subtype of `V1`.
+/// @author sgrekhov22@gmail.com
+
+// SharedOptions=--enable-experiment=inline-class
+
+extension type V1<T>(T _) {}
+
+extension type V2<T extends num>(T _) implements V1<T> {}
```
Comment (eernstg) [inline LanguageFeatures/Extension-types/superinterfaces_of_extension_type_A04_t10.dart:25]: But in this case we _could_ actually use the previous test to get that error. ;-)

File: LanguageFeatures/Extension-types/superinterfaces_of_extension_type_A04_t11.dart [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+/// ...
+/// Assume that an extension type declaration DV named Name has representation
+/// type R, and that the extension type V1 with declaration DV1 is a
+/// superinterface of DV (note that V1 may have some actual type arguments).
+/// Assume that S is the instantiated representation type corresponding to V1. A
+/// compile-time error occurs if R is is neither a subtype of S nor a subtype of
+/// V1.
+///
+/// @description Checks that it is not an error if an extension type has a
+/// representation type `R` and a superinterface 'V1' with a representation type
```
Comment (eernstg) [inline LanguageFeatures/Extension-types/superinterfaces_of_extension_type_A04_t11.dart:17]:
```suggestion
/// representation type `R` and a superinterface `V1` with a representation type
```

**Review by eernstg — COMMENTED** [review]
Looks good, but I think one test needs an adjustment: LanguageFeatures/Extension-types/superinterfaces_of_extension_type_A04_t09.dart.

**Comment by sgrekhov** [comment]
Updated. Please take another look

**Review by eernstg — APPROVED** [review]
LGTM

**Pull request merged**